bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/ModelRepositoryImpl.java

```diff
@@ -202,6 +218,41 @@ private Resource getResource(String name) {
         return resourceSet.getResource(URI.createURI(name), false);
     }
 
+    private String validateModel(String name, InputStream inputStream) throws IOException {
+        // use another resource for validation in order to keep the original one for emergency-removal in case of errors
+        Resource resource = resourceSet.createResource(URI.createURI("tmp_" + name));
+        try {
+            resource.load(inputStream, Collections.EMPTY_MAP);
+            StringBuilder criticalErrors = new StringBuilder();
+            StringBuilder warnings = new StringBuilder();
+
+            if (resource != null && !resource.getContents().isEmpty()) {
+                // Check for syntactical errors
+                for (Diagnostic diagnostic : resource.getErrors()) {
+                    criticalErrors.append(MessageFormat.format("[{0},{1}]: {2}\n", diagnostic.getLine(),
+                            diagnostic.getColumn(), diagnostic.getMessage()));
+                }
+                if (criticalErrors.length() > 0) {
+                    return criticalErrors.toString();
+                }
+
+                // Check for validation errors, but log them only
+                org.eclipse.emf.common.util.Diagnostic diagnostic = Diagnostician.INSTANCE
+                        .validate(resource.getContents().get(0));
+                for (org.eclipse.emf.common.util.Diagnostic d : diagnostic.getChildren()) {
+                    warnings.append(d.getMessage());
+                    warnings.append("\n");
+                }
+                if (warnings.length() > 0) {
+                    logger.warn("Validation error(s) in configuration model '{}':\n {}", name, warnings.toString());
```

Comment:
Shouldn't toString() be called automatically by the logger framework? It make some difference if you call it yourself.
If you call toString yourself, it is executed all the time regardless of the used log level (so, also if the current logger is set to error).